src/lib.rs

```diff
@@ -81,84 +81,153 @@ pub enum ParameterType {
-    pub private_key_path: Option<PathBuf>,
+    #[arg(short = 'k', long = "private_key_path")]
+    private_key_path: Option<PathBuf>,
 
     /// Supply parameters via the command line in <name>=<value> format
-    #[structopt(long = "param")]
-    pub parameters: Vec<String>,
+    #[arg(long = "param")]
+    parameters: Vec<String>,
 }
 
-#[buildstructor::buildstructor]
 impl Opts {
-    #[builder]
-    #[allow(clippy::too_many_arguments)]
-    #[allow(dead_code)]
-    pub fn new(
-        template_path: PathBuf,
-        repository_template_path: Option<PathBuf>,
-        git_ref: Option<String>,
-        project_name: Option<String>,
-        target_dir: Option<PathBuf>,
-        force: Option<bool>,
-        append: Option<bool>,
-        private_key_path: Option<PathBuf>,
-        parameters: Vec<String>,
-    ) -> Opts {
-        Self {
-            template_path,
-            repository_template_path,
-            git_ref,
-            project_name,
-            target_dir,
-            force: force.unwrap_or_default(),
-            append: append.unwrap_or_default(),
-            passphrase_needed: false,
-            private_key_path,
-            parameters,
-        }
+    /// Set the template path for the structure.
+    pub fn template_path(mut self, path: &str) -> Self {
+        let _ = std::mem::replace(&mut self.template_path, path.into());
+        self
+    }
+
+    /// Set the template path inside the repository
+    pub fn repository_template_path(mut self, path: &str) -> Self {
+        let _ = self.repository_template_path.replace(path.into());
+        self
+    }
+
+    /// Set the git reference
+    pub fn git_ref(mut self, gitref: &str) -> Self {
+        let _ = self.git_ref.replace(gitref.to_string());
+        self
+    }
+
+    /// Set the project name
+    pub fn project_name(mut self, name: &str) -> Self {
+        let _ = self.project_name.replace(name.to_string());
+        self
+    }
+
+    /// Set the target directory
+    pub fn target_dir(mut self, target_dir: &str) -> Self {
+        let _ = self.target_dir.replace(target_dir.into());
+        self
+    }
+
+    /// Force generating to the target directory if exists
+    pub fn force(mut self, force: bool) -> Self {
+        self.force = force;
+        self
+    }
+
+    /// Append generating to the target directory if exists
+    pub fn append(mut self, append: bool) -> Self {
+        self.append = append;
+        self
+    }
+
+    /// Is Passphrase needed (prompt user)
+    pub fn passphrase_needed(mut self, needed: bool) -> Self {
+        self.passphrase_needed = needed;
+        self
+    }
+
+    /// Set the private key path
+    pub fn private_key_path(mut self, private_key_path: &str) -> Self {
+        let _ = self.private_key_path.replace(private_key_path.into());
+        self
+    }
+
+    /// Set the parameters (supplied as vec!["key1=value1", "key2=value2"]).
+    pub fn parameters(mut self, params: Vec<&str>) -> Self {
```

Comment:
```suggestion
    pub fn parameters(mut self, params: Vec<String>) -> Self {
```

Reply:
Same logic as above, in general, it's much easier for the user to send anything as `&str` than sending it as `String` we can convert it to string when required.